test/sql/order/test_limit_percent.test

```diff
@@ -0,0 +1,109 @@
+# name: test/sql/order/test_limit_percent.test
+# description: Test LIMIT keyword {% | PERCENT}
+# group: [order]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE test (a INTEGER, b INTEGER);
+
+statement ok
+INSERT INTO test VALUES (11, 22), (12, 21), (13, 22), (14, 32), (15, 52)
+
+# constant limit
+query I
+SELECT a FROM test LIMIT 20 %
+----
+11
+
+query I
+SELECT a FROM test LIMIT 40 PERCENT
+----
+11
+12
+
+# decimal limit
+query I
+SELECT a FROM test LIMIT 66.7%
+----
+11
+12
+13
+
+query I
+SELECT a FROM test LIMIT 101.1 PERCENT
+----
+11
+12
+13
+14
+15
+
+# LIMIT PERCENT with order by
+query I
+SELECT i from generate_series(1, 2000) tbl(i) WHERE (i / 10 * 10 = i) ORDER BY i DESC LIMIT 10%
```

Comment:
Please add the following tests:

* Tests with many values, e.g.

```sql
SELECT * FROM range(1000000) LIMIT 0.0001%
```

* Tests with LIMIT% and OFFSET combined
* LIMIT% with complex types (strings, lists, structs)
* LIMIT 0%
* Test behavior when percentage does not match exactly, e.g. if we have 10 rows, does 25% return 2 or 3 rows? What about 21%? Or 29%? That behavior should probably match Oracle/other systems that support this feature.
* LIMIT% in subqueries, e.g.:
```sql
SELECT * FROM (SELECT * FROM (SELECT * FROM range(1000000) LIMIT 10%) LIMIT 10%) LIMIT 10%;
```
* LIMIT% in set operations, e.g.:
```sql
SELECT * FROM range(10) UNION ALL SELECT * FROM range(10, 20) LIMIT 10%;
```

Reply:
Okay, I will add test case as you said.